frameworks/kafka/tests/auth.py

```diff
@@ -30,7 +30,11 @@ def get_service_principals(service_name: str, realm: str) -> list:
         "kafka-1-broker",
         "kafka-2-broker",
     ]
-    instances = map(lambda task: sdk_hosts.autoip_host(service_name, task), tasks)
+
+    if custom_domain:
```

Comment:
As discussed offline, we could change the function to:
```python
def get_service_principals(domain: str, realm: str) -> list:
    """
    Sets up the appropriate principals needed for a kerberized deployment of HDFS.
    :return: A list of said principals
    """
    primaries = ["kafka", ]

    tasks = [
        "kafka-0-broker",
        "kafka-1-broker",
        "kafka-2-broker",
    ]

    instances = map(lambda task: "{}.{}".format(task, domain), tasks)

    principals = kerberos.generate_principal_list(primaries, instances, realm)
    principals.extend(kerberos.generate_principal_list(USERS, [None, ], realm))

    return principals
```
Were we just pass in the domain and the correct construction is left to the caller.

See my comments on `autoip_host` below.